### PR TITLE
Adds the "Split" circuit component

### PR DESCRIPTION
**Pull request by TheSmallBlue**
## About The Pull Request
Adds a new circuit component called "split". When you give it a string and a separator, it will create a list out of the string, dividing each item by the separator.

For example:
I have the following string: `Boy, do I love the captain!`
If I set the separator to " ", then it will turn the above sentence into the following list list: `"Boy,","do","I","love","the","captain!"`
If I set the separator to ",", then it will turn the above sentence into the following list list: `"Boy"," do I love the captain!"`
And so on and so forth.

## Why It's Good For The Game
Among other things, it allows for retrieving information from strings to later be used in other components. 
For example:
Here is a circuit that sets a person to arrest just by you saying it, no fiddling with the arrest console needed.
image
https://www.youtube.com/watch?v=9B-FVQswipA

## Changelog
:cl:
expansion: Added the "Split" circuitry component.
/:cl:

### CODE REVIEW

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -224,6 +224,7 @@
 		"comp_speech",
 		"comp_speech",
 		"comp_string_contains",
+		"comp_split",
```
Comment (Mothblocks) [inline code/modules/research/techweb/all_nodes.dm:227]: `split` comes before `string_contains`

**Comment by Mothblocks** [comment]
@Watermelon914 You said some stuff about being scared to add list components due to OOM concerns--does this fall in line with that?

If it does, can you provide a clear writing of it on the design doc?

**Comment by Colovorat** [comment]
it would be expansion instead of add right?

**Comment by TheSmallBlue** [comment]
Oh yeah, it would. I'll fix that

**Comment by Watermelon914** [comment]
> @Watermelon914 You said some stuff about being scared to add list components due to OOM concerns--does this fall in line with that?
> 
> If it does, can you provide a clear writing of it on the design doc?

This is fine. It's only a problem if you can somehow add a list to a list because then you can infinitely create lists. This one should be fine.

**Comment by Watermelon914** [comment]
Basically, anything that mutates lists needs to make sure it can't keep inserting lists into itself, though this could probably just be some hardcoded check but it's easy to get wrong and easy to let players crash the server if you do get it wrong.

**Pull request merged**